src/helpers/radspec.js

```diff
@@ -0,0 +1,64 @@
+const ABI = require('web3-eth-abi')
+const { keccak256 } = require('web3-utils')
+
+const getSig = (fn) =>
+  keccak256(fn).substr(0, 10)
+
+// Convert from the knownFunctions data format into the needed format
+// Input: { "signature(type1,type2)": "Its radspec string", ... }
+// Output: { "0xabcdef12": { "sig": "signature(type1,type2)", "source": "Its radspec string" }, ...}
+const processFunctions = (functions) => (
+  Object.keys(functions).reduce(
+    (acc, key) => (
+      {
+        [getSig(key)]: { source: functions[key], sig: key },
+        ...acc
+      }
+  ), {})
+)
+
+module.exports = (eth) =>
+  /**
+   * Evaluate a known radspec string
+   *
+   * @param {address} addr The target address of the call
+   * @param {bytes} data The calldata of the call
+   * @return {Promise<radspec/evaluator/TypedValue>}
+   */
+  async (addr, data) => {
+    // lazily import radspec to avoid a dependency cycle
+    const { evaluateRaw } = require('../index')
```

Comment:
.__. not sure how well this is going to fly with builders like parcel. Will need to check.

Reply:
🤷‍♀️ cleaner way I could think of that didn't require a major re-architecture